File: src/minimal_palindrome/minimal_palindrome.cpp
#include <algorithm>
#include <iostream>
#include <string>

using namespace std;

int main(void) {
    string s;
    cin >> s;
    int cnt[26] = {0};
    for (auto c : s) ++cnt[c - 'a'];
    string half, mid;
    for (int i = 0; i < 26; ++i) {
        char c = 'a' + i;
        half += string(cnt[i] / 2, c);
        if (cnt[i] % 2 && mid.empty()) mid += c;
    }

    string half2 = half;
    reverse(half2.begin(), half2.end());
    string ans = half + mid + half2;
    cout << ans << endl;
    return 0;
}

File: src/minimal_palindrome/minimal_palindrome.py
import collections

s = input()

cnt = collections.Counter(s)

half = ""
mid = ""

for c in sorted(cnt.keys()):
    half += c * (cnt[c] // 2)
    if cnt[c] % 2 and not mid:
        mid = c

ans = half + mid + half[::-1]
print(ans)